### DISCUSSION TITLE
Asynchronous Circuit Implementation

### DISCUSSION THREAD

**Discussion by mrberman87** [SpinalHDL/SpinalHDL#1619] [Q&A]
Does anybody have any suggestions on how to implement an asyncronous design in Spinal?

I am looking at implementing a PFD ( see page 19 of [here](https://people.engr.tamu.edu/spalermo/ecen620/lecture04_ee620_phase_detectors.pdf) ).

Is this do-able?  I would have to do a clock domain for each of the two clocked flip-flops, but then how would I route the reset lines?  As well as moving the up/down pulses into a system core clock domain, with guaranteed visibility, even for very narrow pulses.

**Comment by Dolu1990** [comment]
Hooo tricky XD I don't know enough about this topic.
You may prefer using blackboxes for the primitives ?
